File: 0283-move-zeroes/0283-move-zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int nonZeroIndex = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] != 0) {
            nums[nonZeroIndex++] = nums[i];
        }
    }
    for (int i = nonZeroIndex; i < numsSize; i++) {
        nums[i] = 0;
    }
}

void printArray(int* nums, int numsSize) {
    printf("[");
    for (int i = 0; i < numsSize; i++) {
        printf("%d", nums[i]);
        if (i < numsSize - 1) {
            printf(", ");
        }
    }
    printf("]\n");
}

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n=nums.length,ind=0;
        for(int i=0;i<n;i++){
            if(nums[i]!=0)nums[ind++]=nums[i];
        }
        for(int i=ind;i<n;i++)nums[i]=0;
    }
}